examples/ipython/tutorial_algebra.ipynb

```diff
@@ -0,0 +1,672 @@
+   "outputs": [
+    {
+     "data": {
+      "text/latex": [
+       "$\\displaystyle \n",
+       "\\begin{align}\n",
+       "    a &= a^{x} \\boldsymbol{e}_{x} + a^{y} \\boldsymbol{e}_{y} + a^{z} \\boldsymbol{e}_{z} \\\\\n",
+       "    b &= b^{x} \\boldsymbol{e}_{x} + b^{y} \\boldsymbol{e}_{y} + b^{z} \\boldsymbol{e}_{z}\n",
+       "\\end{align}\n",
+       "$"
+      ],
+      "text/plain": [
+       "<IPython.core.display.Math object>"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "a = o3d.mv('a','vector')\n",
+    "b = o3d.mv('b','vector')\n",
+    "Math(fr'''\n",
+    "\\begin{{align}}\n",
+    "    a &= {latex(a)} \\\\\n",
+    "    b &= {latex(b)}\n",
+    "\\end{{align}}\n",
+    "''')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/latex": [
+       "$\\displaystyle \n",
+       "\\begin{align}\n",
+       "    a+b         &= \\left ( a^{x} + b^{x}\\right ) \\boldsymbol{e}_{x} + \\left ( a^{y} + b^{y}\\right ) \\boldsymbol{e}_{y} + \\left ( a^{z} + b^{z}\\right ) \\boldsymbol{e}_{z} \\\\\n",
+       "    a-b         &= \\left ( a^{x} - b^{x}\\right ) \\boldsymbol{e}_{x} + \\left ( a^{y} - b^{y}\\right ) \\boldsymbol{e}_{y} + \\left ( a^{z} - b^{z}\\right ) \\boldsymbol{e}_{z} \\\\\n",
+       "    ab          &= \\left ( a^{x} b^{x} + a^{y} b^{y} + a^{z} b^{z}\\right )  + \\left ( a^{x} b^{y} - a^{y} b^{x}\\right ) \\boldsymbol{e}_{x}\\wedge \\boldsymbol{e}_{y} + \\left ( a^{x} b^{z} - a^{z} b^{x}\\right ) \\boldsymbol{e}_{x}\\wedge \\boldsymbol{e}_{z} + \\left ( a^{y} b^{z} - a^{z} b^{y}\\right ) \\boldsymbol{e}_{y}\\wedge \\boldsymbol{e}_{z} \\\\\n",
+       "    a\\cdot b    &= a^{x} b^{x} + a^{y} b^{y} + a^{z} b^{z} \\\\\n",
+       "    a \\rfloor b &= a^{x} b^{x} + a^{y} b^{y} + a^{z} b^{z} \\\\\n",
+       "    a \\lfloor b &= a^{x} b^{x} + a^{y} b^{y} + a^{z} b^{z} \\\\\n",
+       "    a\\wedge b   &= \\left ( a^{x} b^{y} - a^{y} b^{x}\\right ) \\boldsymbol{e}_{x}\\wedge \\boldsymbol{e}_{y} + \\left ( a^{x} b^{z} - a^{z} b^{x}\\right ) \\boldsymbol{e}_{x}\\wedge \\boldsymbol{e}_{z} + \\left ( a^{y} b^{z} - a^{z} b^{y}\\right ) \\boldsymbol{e}_{y}\\wedge \\boldsymbol{e}_{z}\n",
+       "\\end{align}\n",
+       "$"
+      ],
+      "text/plain": [
+       "<IPython.core.display.Math object>"
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "Math(fr'''\n",
+    "\\begin{{align}}\n",
+    "    a+b         &= {latex(a+b)} \\\\\n",
+    "    a-b         &= {latex(a-b)} \\\\\n",
+    "    ab          &= {latex(a*b)} \\\\\n",
+    "    a\\cdot b    &= {latex(a|b)} \\\\\n",
+    "    a \\rfloor b &= {latex(a<b)} \\\\\n",
+    "    a \\lfloor b &= {latex(a>b)} \\\\\n",
+    "    a\\wedge b   &= {latex(a^b)}\n",
+    "\\end{{align}}\n",
+    "''')"
```

Comment:
Using `Math` and `latex` in this way is a little clumsy, but it's not completely unreadable, and the syntax highlighting in the notebook makes it more palatable (see reviewNB)

Reply:
<p>Another option here would be to use something like</p><pre class="ql-syntax" spellcheck="false">display(
    (a+b).Fmt(title='a+b'),
    (a-b).Fmt(title='a-b'),
    (a*b).Fmt(title='ab'),
    (a|b).Fmt(title=r'a \cdot b'),
    ...
)
</pre><p>Which is perhaps a little more readable, but:</p><ul><li>Looks less pretty in the output</li><li>Does not appear in the <code>Out[n]</code>  cell any more.</li><li>Uses <code>Fmt</code>,  which we were trying not to mention till later</li><li>Only works on <code>mv</code>  objects, which means we end up having to switch between different notations</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pygae/galgebra/pull/375/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Reply:
Without `__str__`, it's really a bit verbose, calling `latex()` repeatedly is annoying.

Reply:
So we don't have support for `str()`, `repr()` now?

Reply:
While I'm not happy with the old implementation and happy with the new implementation, but I do enjoy the old stateful way to configure the format globally to save such verbosity.

Reply:
What are your thoughts on the `display` variant?

Reply:
I believe that is still compatible with current implementation, right?

Reply:
If you mean "does it work today", yes. The downside is mainly that it does not align the equals signs.

Reply:
> What are your thoughts on the `display` variant?

Didn't like it either, I used it, but if I recall correctly they result in multiple separate MathJax units and very cumbersome.

Reply:
The upside is that it produces nice output in a shell that does not support latex.

Reply:
Multiple mathjax outputs is how the proposed `gprint` would work anyway.

Just to be clear, I'm referring to https://github.com/pygae/galgebra/pull/375#discussion_r428761611

Reply:
> The upside is that it produces nice output in a shell that does not support latex.

But can a user explicitly asks for auto call on `latex()` on `str()` or `repr()`.

Reply:
I don't see any reason they'd want it for `repr`.

As for `str`, the use-cases I can think of are:
* Passing to `print`. On master this still prints latex if you call `Format`, but it does not and has never worked in jupyter.
* Concatenating into a latex string to display with `Latex` or `Math`, like `Math('s = ' + str(x))`, or `Math('s = %s' % x)`

Note that for `Math(f's = {x}')` to work, we actually don't need `__str__` at all, we can overload `__format__`.

Reply:
> Multiple mathjax outputs is how the proposed `gprint` would work anyway.

I guess that would do.

Reply:
Whatever we come up with, it might make sense to contribute it back to sympy - none of these problems are really galgebra's

Reply:
Yeah